test/export/test_export.py

```diff
@@ -1505,5 +1506,67 @@ def forward(self, arg0_1):
     zeros = torch.ops.aten.zeros.default([_local_scalar_dense], device = device(type='cpu'), pin_memory = False);  _local_scalar_dense = None
     return (zeros,)""")
 
+    def test_non_arg_name_dynamic_shapes_api(self):
+        def foo(a, b):
+            return a.sum() + b.sum()
+
+        dim = torch.export.Dim("dim")
+        ep = torch.export.export(foo, (torch.randn(4, 4), torch.randn(4, 4)), dynamic_shapes=(None, {0: dim}))
+
+        test_inp = (torch.randn(4, 4), torch.randn(7, 4))
+        self.assertEqual(ep(*test_inp), foo(*test_inp))
+
+        ep_v2 = torch.export.export(foo, (torch.randn(4, 4), torch.randn(4, 4)), dynamic_shapes=(None, None))
+        with self.assertRaisesRegex(RuntimeError, "Input arg1_1.shape\[0\] is specialized at 4"):
+            ep_v2(*test_inp)
+
+    def test_non_arg_name_dynamic_shapes_api_with_kwarg(self):
+        def foo(a, b, kw1, kw2):
+            return a.sum() + b.sum() + kw1.sum() - kw2.sum()
+
+        dim = torch.export.Dim("dim")
+        dim_2 = torch.export.Dim("dim_2")
+        ep = torch.export.export(
+            foo,
+            (torch.randn(4, 4), torch.randn(4, 4)),
+            {"kw2": torch.ones(4, 4), "kw1": torch.zeros(4, 4)},
+            dynamic_shapes=(None, {0: dim}, {0: dim_2}, None))
```

Comment:
The naming in this test is confusing because `dim_2` is describing `kw1`. Mind renaming and adding a comment?